src/vtt-segment-loader.js

```diff
@@ -380,14 +382,47 @@ export default class VTTSegmentLoader extends SegmentLoader {
     this.handleAppendsDone_();
   }
 
-  handleData_() {
-    // noop as we shouldn't be getting video/audio data captions
-    // that we do not support here.
+  handleData_(simpleSegment, result) {
+    const isFmp4VttSegment = simpleSegment.type === 'vtt' && result.type === 'text';
```

Comment:
Could errors possibly be thrown here if `simpleSegment` or `result` are undefined? It might be worth to have a check here

Reply:
Good call, will make this check a bit more robust.